jsonParser/semi-automatic/parse_election_results.py

```diff
@@ -0,0 +1,187 @@
+import csv
+import json
+import click
+import os
+
+@click.command()
+@click.option('--input', '-i', help='Input CSV file', required=True)
+@click.option('--output', '-o', help='Output JSON file', required=True)
+def main(input:str, output:str):
+    check_file_exists(input)
+    
+    # Read the CSV file
+    with open(input, 'r', encoding='utf-8-sig') as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=';')
+        rows_list = create_list(csv_reader)
+    
+    # Create the dictionary that will contain the data
+    data = {
+        "schede": {
+            "bianche": {
+                "totali": 0,
+                "seggio_n_telematico": 0
+            },
+            "nulle": {
+                "totali": 0
+            },
+            "contestate": {
+                "totali": 0
+            }
+        },
+        "liste": [],
+        "eletti": [],
+        "non_eletti": [],
+    }
+    
+    rows_list = get_name_and_seats(rows_list, data)
+    rows_list = get_list_information(rows_list, data)
+    rows_list = get_votation_information(rows_list, data)
+    get_candidates_information(rows_list, data)
+
+    # Convert data to JSON format
+    json_output = json.dumps(data, indent=4, ensure_ascii=False)
+
+    # Save the JSON data to a file
+    with open(output, 'w', encoding='utf-8') as json_file:
+        json_file.write(json_output)
+        
+def check_file_exists(file_path:str) -> None:
+    if not os.path.isfile(file_path):
+        raise FileNotFoundError(f"File '{file_path}' not found")
+
+def create_list(csv_reader:list) -> list:
+    """Create a list of rows from the CSV file
+
+    Args:
+        csv_reader (list): The CSV reader
+
+    Returns:
+        list: The list of rows of the CSV file
+    """
+    rows_list = []
+    for row in csv_reader:
+        # Replace \xa0 with a space in the entire row
+        row = [cell.replace('\xa0', ' ') for cell in row]
+        rows_list.append(row)
+    return rows_list
+
+def get_name_and_seats(rows_list:list, data:dict) -> list:
+    """Get the name of the department and the number of seats to be assigned
+
+    Args:
+        rows_list (list): The list of rows of the CSV file
+        data (dict): The dictionary that will contain the data
+    
+    Returns:
+        list: The list of rows of the CSV file
+    """
+    row = rows_list[0]
+    data["dipartimento"] = str(row[0])
+    row = rows_list[1]
+    data["seggi_da_assegnare"] = row[1]
+    rows_list = rows_list[4:]
+    return rows_list
+    
+def get_list_information(rows_list:list, data:dict) -> list:

```

Comment:
I strongly suggest to put a space between variable name and type.
In the future we will the auto-formatter `black` in all projects to fix the indentations and spaces automatically
```suggestion
def get_list_information(rows_list: list, data: dict) -> list:

```